scipy/stats/_stats_py.py

```diff
@@ -318,14 +318,15 @@ def gmean(a, axis=0, dtype=None, weights=None):
     """
     if not isinstance(a, np.ndarray):
```

Comment:
If we're changing the logic here, why don't we get rid of this? I think we only need to convert to an array or masked array if `dtype` is specified. Otherwise, we can go straight to taking the log of `a`, right?